core/src/apps/cardano/layout.py

```diff
@@ -123,7 +123,7 @@ async def show_warning_tx_different_staking_account(
 
     page2 = Text("Confirm transaction", ui.ICON_SEND, ui.GREEN)
     page2.normal("Staking account:")
-    page2.bold(address_n_to_str(staking_account_path))
+    page2.bold("#%d" % to_account_number(staking_account_path))
```

Comment:
what happens if a nonstandard path is used?

Reply:
I think `to_account_number` should contain some sanity checks and return None or raise when they fail. In that case we can still show `address_n_to_str` instead.

Reply:
we refactored it to `format_account_number()` and it enforces the path length to be at least 3, otherwise it raises a ValueError as that is an obviously malformed input, without doing further checks which should be handled outside of the logic to format the account number itself